src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -26,8 +26,8 @@ public DiagnosticsHandler(HttpMessageHandler innerHandler) : base(innerHandler)
 
         internal static bool IsEnabled()
         {
-            //check if someone listens to HttpHandlerDiagnosticListener
-            return s_diagnosticListener.IsEnabled();
+            //check if someone listens to HttpHandlerDiagnosticListener of if there's a current activity
+            return Activity.Current != null || s_diagnosticListener.IsEnabled();
```

Comment:
This check isn't as fast as I originally thought, but I haven't measured the overhead (plus it's about to make an HTTP request 😄 )

Reply:
Yeah last I looked, it wasn't all that trivial to dig out an async local. Perf-impact aside it looked fine to me. I assume the various opt-out mechanisms you mentioned in the issue are still forthcoming.

Reply:
Yep. Some of those require public API (which means an API review), so I wanted to make a draft of the core before going there. Also I need to write tests.